### PR TITLE
Feature/tab name change

### LINKED ISSUE #27: change name of the tab
Something like Twitter| Focus Mode & disable notifications

Screenshot 2024-04-16 at 10 14 35 PM

### PR DESCRIPTION
**Pull request by 02-ahmed**
Addresses the feature request of adding "Focus Mode & Disable Notifications" to the tab name of blocked sites.

# Closes #27

### CODE REVIEW

File: my-extension/src/components/Restrict.tsx [hunk 1]
```diff
@@ -30,19 +30,20 @@ function Restrict() {
       <div
         style={{
           display: "flex",
+          flexDirection:"column", //added flex direction column because i moved the random message inside this div
```
Comment (pritipsingh) [inline my-extension/src/components/Restrict.tsx]: we don't need the comment here

File: my-extension/src/components/Restrict.tsx [hunk 2]
```diff
@@ -30,19 +30,20 @@ function Restrict() {
...
           justifyContent: "center",
           alignItems: "center",
         }}
       >
         <img
           src="https://cdn.dribbble.com/users/285475/screenshots/2083086/dribbble_1.gif"
           alt="restrict gif"
+          style={{maxHeight:"500px"}} //set max height of gif to 500px so it doesn't push the random message out of the view height
```
Comment (pritipsingh) [inline my-extension/src/components/Restrict.tsx]: again comment not needed

File: my-extension/src/components/Restrict.tsx [hunk 3]
```diff
@@ -30,19 +30,20 @@ function Restrict() {
...
         />
-      </div>
-
-      <h3 style={{ textAlign: "center", fontSize: "30px", color: "black" }}>
+        <h3 style={{ textAlign: "center", fontSize: "30px", color: "black" }}>
         {randomMessage}
-      </h3>
+        </h3> {/* moved the h3 tag inside the div that contains the image to ensure that the random message is always underneath the gif */}
```
Comment (pritipsingh) [inline my-extension/src/components/Restrict.tsx]: this too

File: my-extension/src/components/WebsiteEditPage.tsx [hunk 4]
```diff
@@ -17,23 +17,36 @@ export default function WebsiteEditPage() {
...
+      // If savedItems exists and has items, use it
+      setSlreadyBlockedWebsitesSet(savedItems);
+      } else {
+      // If savedItems is null, empty, or parsing failed
+      if (hasOpened) {
+      // If the app has been opened before, set to an empty array
+      setSlreadyBlockedWebsitesSet([]);
+      } else {
+      // If the app has not been opened before, use the hardcoded list and mark as opened
+      setSlreadyBlockedWebsitesSet(alreadyBlockedWebsites); // Assuming 'alreadyBlockedWebsites' is an array
```
Comment (pritipsingh) [inline my-extension/src/components/WebsiteEditPage.tsx:45]: these comments are fine

File: my-extension/src/components/WebsiteEditPage.tsx [hunk 5]
```diff
@@ -17,23 +17,36 @@ export default function WebsiteEditPage() {
...
+      setHasOpened(true); // This marks the initial setup as done
+      }
       }
-    }, []);
+      }, []); //this was different from the code in the deployment so copied the deployed code back which caused the edit websites to block to start working again
```
Comment (pritipsingh) [inline my-extension/src/components/WebsiteEditPage.tsx]: not needed please remove

File: my-extension/src/content/index.tsx [hunk 6]
```diff
@@ -27,10 +31,14 @@ chrome.storage.local.get(["websitesToBlock"], function (result) {
...
     const div = document.createElement("div");
+    div.id = 'restrict-overlay'
     document.body.style.overflow = "hidden";
     document.body.appendChild(div);
     ReactDOM.render(<Restrict />, div);
   }
+
+  document.title += ` | Focus Mode & Disable Notifications` //appended the text to the existing tab name by using tempelate literals to ensure the white spaces take effect
 });
+//content script was running multiple times so added code to check if the restrict component is already rendered before adding a new one
```
Comment (pritipsingh) [inline my-extension/src/content/index.tsx]: not needed

File: my-extension/src/content/index.tsx [hunk 7]
```diff
@@ -27,10 +31,14 @@ chrome.storage.local.get(["websitesToBlock"], function (result) {
...
-  if (restricExist) {
+  if (restricExist && !document.getElementById('restrict-overlay')) {
     const div = document.createElement("div");
+    div.id = 'restrict-overlay'
     document.body.style.overflow = "hidden";
     document.body.appendChild(div);
     ReactDOM.render(<Restrict />, div);
   }
+
+  document.title += ` | Focus Mode & Disable Notifications` //appended the text to the existing tab name by using tempelate literals to ensure the white spaces take effect
```
Comment (pritipsingh) [inline my-extension/src/content/index.tsx]: this is fine but make it shorter

File: my-extension/src/components/Restrict.tsx (hunk 2 above)
Comment (pritipsingh) [inline my-extension/src/components/Restrict.tsx]: any reason to particularly use px and not rem or vh

**Review by pritipsingh — CHANGES_REQUESTED** [review]

File: my-extension/src/components/Restrict.tsx (hunk 2 above)
Comment (02-ahmed) [reply my-extension/src/components/Restrict.tsx]: no particular reason
changed it to use vh

**Comment by pritipsingh** [comment]
Thanks for the contributions @02-ahmed ! If you ever have your sponsor button enabled, feel free to email me. Would love to support you in your OS journey.

**Comment by 02-ahmed** [comment]
Thank You Priti

Glad to have been able contribute
I would like to reach out if i need guidance or if i encounter any
challenges I need help with
Where can i reach you more easily ?

Thanks again

**Comment by 02-ahmed** [comment]
Hello Pritipriya,

Trust you are well
I noticed that my pull requests that you merged a while back have not been
deployed. So the bugs that are supposed to be resolved are still there when
you install the extension from the chrome store.
I have noticed another bug that I would like to work on so if you can check
the deployment issue out that would be great.
Is there a social media you're comfortable with? so I can be reaching out
more easily

Also, my github sponsor button is enabled now

Hoping to hear from you soon

Best regards,
Ahmed

**Comment by 02-ahmed** [comment]
Hello Pritipriya,

Trust you are well
I noticed that my pull requests that you merged a while back have not been
deployed. So the bugs that are supposed to be resolved are still there when
you install the extension from the chrome store.
I have noticed another bug that I would like to work on so if you can check
the deployment issue out that would be great

Also, my github sponsor button is enabled now

Hoping to hear from you soon

Best regards,
Ahmed

**Pull request merged**